### PR TITLE
chore: use github API library list in README.rst

### LINKED ISSUE #10533: Synthesis failed for google-cloud-python
Hello! Autosynth couldn't regenerate google-cloud-python. :broken_heart:

Here's the output from running `synth.py`:

```
2021-01-30 05:13:14,117 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/google-cloud-python
2021-01-30 05:13:19,330 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2021-01-30 05:13:19,332 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2021-01-30 05:13:19,335 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2021-01-30 05:13:19,337 autosynth [DEBUG] > Running: git config push.default simple
2021-01-30 05:13:19,340 autosynth [DEBUG] > Running: git branch -f autosynth
2021-01-30 05:13:19,343 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-01-30 05:13:19,350 autosynth [DEBUG] > Running: git rev-parse --show-toplevel
2021-01-30 05:13:19,353 autosynth [DEBUG] > Running: git log -1 --pretty=%H
2021-01-30 05:13:19,356 autosynth [DEBUG] > Running: git remote get-url origin
2021-01-30 05:13:19,359 autosynth [DEBUG] > Running: git checkout e789e3d850b2cbd60816a26a33b3e538b60bdc3d
Note: checking out 'e789e3d850b2cbd60816a26a33b3e538b60bdc3d'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at e789e3d85 chore: start tracking obsolete files (#10531)
2021-01-30 05:13:19,363 autosynth [DEBUG] > Running: git branch -f autosynth-0
2021-01-30 05:13:19,366 autosynth [DEBUG] > Running: git checkout autosynth-0
Switched to branch 'autosynth-0'
2021-01-30 05:13:19,369 autosynth [INFO] > Running synthtool
2021-01-30 05:13:19,370 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2021-01-30 05:13:19,370 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-cloud-python/0/sponge_log.log
2021-01-30 05:13:19,371 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2021-01-30 05:13:19,590 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-cloud-python/synth.py.
On branch autosynth-0
nothing to commit, working tree clean
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-cloud-python/synth.py", line 149, in <module>
    clients = sorted(all_clients())
  File "/home/kbuilder/.cache/synthtool/google-cloud-python/synth.py", line 142, in all_clients
    for repo in response.json()["repos"]
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/requests/models.py", line 900, in json
    return complexjson.loads(self.text, **kwargs)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/json/decoder.py", line 342, in decode
    raise JSONDecodeError("Extra data", s, end)
json.decoder.JSONDecodeError: Extra data: line 1 column 4 (char 3)
2021-01-30 05:13:19,907 autosynth [ERROR] > Synthesis failed
2021-01-30 05:13:19,907 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at e789e3d85 chore: start tracking obsolete files (#10531)
2021-01-30 05:13:19,912 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-01-30 05:13:19,917 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/72ece7fa-e470-41ab-ad07-6a74d5284409/targets/github%2Fsynthtool;config=default/tests;query=google-cloud-python;failed=false).

### PR DESCRIPTION
**Pull request by parthea**
Fixes #10533 🦕

Use a github action instead of autosynth to update the API library list.

### CODE REVIEW

File: synth.py [hunk 1]
```diff
@@ -129,19 +129,25 @@ def client_for_repo(repo_slug) -> Optional[CloudClient]:
...
 def all_clients() -> List[CloudClient]:
-    response = requests.get(REPO_LIST_JSON)
-    clients = [
-        client_for_repo(repo["repo"])
-        for repo in response.json()["repos"]
-        if allowed_repo(repo)
-    ]
+    clients = []
+    page_number = 1
+    while page_number < 10:
```
Comment (busunkim96) [inline synth.py:144]: Can the link header be used to terminate the loop instead? https://2.python-requests.org/en/master/user/advanced/#id14

**Review by busunkim96 — COMMENTED** [review]
Thanks for updating this!

Are there any issues with GitHub API quota?

**Comment by busunkim96** [comment]
@parthea RE: API quota, I'd also be okay to merge this and revisit if there are issues. :) I know the list is pretty out of date at this point.

File: synth.py (hunk 1 above)
Comment (parthea) [reply synth.py]: Done.

**Comment by parthea** [comment]
I'm going to merge this based on the last comment. I've also migrated from autosynth to github action. I copied the github action from https://github.com/googleapis/google-cloud-node/blob/master/.github/workflows/update-api-list.yaml and modified it slightly.

**Pull request merged**